Index: src/test/java/edu/virginia/sde/hw5/DatabaseDriverTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/edu/virginia/sde/hw5/DatabaseDriverTest.java b/src/test/java/edu/virginia/sde/hw5/DatabaseDriverTest.java
new file mode 100644
--- /dev/null	(date 1713414275988)
+++ b/src/test/java/edu/virginia/sde/hw5/DatabaseDriverTest.java	(date 1713414275988)
@@ -0,0 +1,76 @@
+package edu.virginia.sde.hw5;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class DatabaseDriverTest {
+
+    @Test
+    void connect() {
+    }
+
+    @Test
+    void commit() {
+    }
+
+    @Test
+    void rollback() {
+    }
+
+    @Test
+    void disconnect() {
+    }
+
+    @Test
+    void createTables() {
+    }
+
+    @Test
+    void addStops() {
+    }
+
+    @Test
+    void getAllStops() {
+    }
+
+    @Test
+    void getStopById() {
+    }
+
+    @Test
+    void getStopsByName() {
+    }
+
+    @Test
+    void addBusLines() {
+    }
+
+    @Test
+    void getBusLines() {
+    }
+
+    @Test
+    void getBusLinesById() {
+    }
+
+    @Test
+    void getBusLineByLongName() {
+    }
+
+    @Test
+    void getBusLineByShortName() {
+    }
+
+    @Test
+    void getBusLinesByStop() {
+    }
+
+    @Test
+    void getRouteForBusLine() {
+    }
+
+    @Test
+    void clearTables() {
+    }
+}
\ No newline at end of file
Index: src/test/java/edu/virginia/sde/hw5/BusLineServiceTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/edu/virginia/sde/hw5/BusLineServiceTest.java b/src/test/java/edu/virginia/sde/hw5/BusLineServiceTest.java
new file mode 100644
--- /dev/null	(date 1713414275989)
+++ b/src/test/java/edu/virginia/sde/hw5/BusLineServiceTest.java	(date 1713414275989)
@@ -0,0 +1,18 @@
+package edu.virginia.sde.hw5;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BusLineServiceTest {
+
+
+    @Test
+    void getClosestStop() {
+
+    }
+
+    @Test
+    void getRecommendedBusLine() {
+    }
+}
\ No newline at end of file
Index: src/test/java/edu/virginia/sde/hw5/BusLineReaderTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/edu/virginia/sde/hw5/BusLineReaderTest.java b/src/test/java/edu/virginia/sde/hw5/BusLineReaderTest.java
new file mode 100644
--- /dev/null	(date 1713414275986)
+++ b/src/test/java/edu/virginia/sde/hw5/BusLineReaderTest.java	(date 1713414275986)
@@ -0,0 +1,57 @@
+package edu.virginia.sde.hw5;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.net.MalformedURLException;
+import java.util.HashSet;
+import java.util.Set;
+import static org.mockito.Mockito.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+class BusLineReaderTest {
+
+    @Mock
+    private URL stops;
+
+    @Mock
+    private URL lines;
+
+    @Mock
+    private Configuration configuration;
+
+    private BusLineReader busLineReader;
+
+    @BeforeEach
+    void setUp() throws MalformedURLException {
+        stops = new URL("https://www.cs.virginia.edu/~pm8fc/busses/stops.ison");
+        lines = new URL("https://www.cs.virginia.edu/~pm8fc/busses/lines.ison");
+        when(configuration.getBusLinesURL()).thenReturn(lines);
+        when(configuration.getBusStopsURL()).thenReturn(stops);
+        busLineReader = new BusLineReader(configuration);
+    }
+
+    @Test
+    void getBusLines() {
+//        List<Busline> buslineList;
+//        buslineList = busLineReader.getBusLines();
+    }
+}
\ No newline at end of file
Index: src/test/java/edu/virginia/sde/hw5/StopReaderTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/edu/virginia/sde/hw5/StopReaderTest.java b/src/test/java/edu/virginia/sde/hw5/StopReaderTest.java
new file mode 100644
--- /dev/null	(date 1713414275987)
+++ b/src/test/java/edu/virginia/sde/hw5/StopReaderTest.java	(date 1713414275987)
@@ -0,0 +1,12 @@
+package edu.virginia.sde.hw5;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class StopReaderTest {
+
+    @Test
+    void getStops() {
+    }
+}
\ No newline at end of file
Index: README.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>[![Review Assignment Due Date](https://classroom.github.com/assets/deadline-readme-button-24ddc0f5d75046c5622901739e7c5dd533143b0c8e959d652212380cedb1ea36.svg)](https://classroom.github.com/a/syjMWJ6t)\n# Homework 5 - Data Layer\n\n## Authors\n1) Enhui Li, cmk8wq, [lexiliiii]\n2) Casey Shi, nqs9yv, [CaseyyyyS]\n3) Ziqian Zhao, uja2mg, [Sherryzhao23]\n\n\n## Contributions\n\nList the primary contributions of each author. It is recommended to update this with your contributions after each coding session.:\n\n### [Enhui Li]\n\n* Implemented PART_3 and PART_4\n* Helped check PART_1 and PART_2\n* Wrote corresponding testings for PART_2 and PART_3\n\n### [Casey Shi]\n\n* Implemented PART_1\n* Implemented PART_5\n* Wrote corresponding testings for PART_5\n  \n### [Ziqian Zhao]\n\n* Implemented PART_2\n* Helped check PART_4\n* Wrote corresponding testings for PART_5\n\n## Issues\n\nNo issues are being found when submitted.\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/README.md b/README.md
--- a/README.md	(revision b633ae359cef9cc50a16cfaaeaabadc86718d17d)
+++ b/README.md	(date 1713414382895)
@@ -14,7 +14,7 @@
 ### [Enhui Li]
 
 * Implemented PART_3 and PART_4
-* Helped check PART_1 and PART_2
+* Implemented PART_1 and PART_2 and PART_5
 * Wrote corresponding testings for PART_2 and PART_3
 
 ### [Casey Shi]
